diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/KdocComments.kt

```diff
@@ -45,7 +62,148 @@ class KdocComments(private val configRules: List<RulesConfig>) : Rule("kdoc-comm
             when (node.elementType) {
                 FILE -> checkTopLevelDoc(node)
                 CLASS -> checkClassElements(node)
+                VALUE_PARAMETER -> checkValueParameter(node)
+            }
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun checkValueParameter(node: ASTNode) {
+        if (node.parents().none { it.elementType == PRIMARY_CONSTRUCTOR } ||
+                !(node.hasChildOfType(VAL_KEYWORD) || node.hasChildOfType(VAR_KEYWORD))) return
+        val prevComment = if (node.treePrev.elementType == WHITE_SPACE &&
+                (node.treePrev.treePrev.elementType == EOL_COMMENT ||
+                        node.treePrev.treePrev.elementType == BLOCK_COMMENT)) {
+            node.treePrev.treePrev
+        } else if (node.hasChildOfType(KDOC)) {
+            node.findChildByType(KDOC)!!
+        } else if (node.treePrev.elementType == BLOCK_COMMENT || node.treePrev.elementType == EOL_COMMENT) {
```

Comment:
This is the case when comment is immediately before the property? I think the only possibility is something like
```kotlin
constructor(/*comment*/val field: Type)
```
(without space after `*/`) and `EOL_COMMENT` can't occur here. Please check once more and add the corresponding test case.